build.py

```diff
@@ -550,10 +556,13 @@ def clean(args: argparse.Namespace, env: dict[str, str]):
     """
     Clean the build output.
     """
-    log.info("Cleaning targets")
-    cmd_args = [str(args.cmake), "--build", str(args.build_dir), "--config", args.config, "--target", "clean"]
+    log.info("Cleaning cmake targets")
+    cmd_args = [str(args.cmake_path), "--build", str(args.build_dir), "--config", args.config, "--target", "clean"]
```

Comment:
Do we need to clean if we are deleting the build directory? If we are deleting the build directory, it offers no advantage to the user since doing

```
rm -rf build/
```

or 

```
rmdir /s /q "build"
```

is probably easier for the user. Maybe we can just let clean be the same as cleaning the cmake output dir.